src/yupSchema.js

```diff
@@ -74,9 +78,58 @@ function getRequiredErrorMessage(inputType, { inlineError, configError }) {
 
 const getJsonTypeInArray = (jsonType) =>
   Array.isArray(jsonType)
-    ? jsonType.find((val) => val !== 'null') // eg ["string", "null"] // optional fields - get the lead type.
+    ? jsonType.find((val) => val !== 'null') // eg ["string", "null"] // optional fields - get the head type.
     : jsonType; // eg "string"
 
+const getOptionsAllowed = (field) => {
+  const onlyValues = field.options?.map((option) => option.value);
+  const optionsBroken = [...onlyValues, '']; /*[1]*/
+
+  if (field.required) {
+    return optionsBroken;
+  }
+
+  const isOptionalWithNull =
+    Array.isArray(field.jsonType) &&
+    // @TODO should also check the "oneOf" directly looking for "null" option
+    // but we don't have direct access at this point. Otherwise
+    // the JSON Schema validator will fail because setting jsonType is not enough.
+    field.jsonType.includes('null');
+  if (isOptionalWithNull) {
+    return [...optionsBroken, null];
```

Comment:
FYI I've created a dev release `0.4.1-dev.20230702181843` which includes this and #23. I'm testing it in our [internal repo](https://gitlab.com/remote-com/employ-starbase/dragon/-/merge_requests/17915)

Reply:
After all this `null` will always be necessary similar to `""`, otherwise our empty fields: Look at the [Radio demo](https://github.com/remoteoss/json-schema-form/assets/14869087/e431bea4-33bb-42fa-a3dd-589fe13fc9e2). (I spot this through our internal Storybook tests.

Changing the code (and code comment) now.

Reply:
Update: Done: [93fbd2e](https://github.com/remoteoss/json-schema-form/pull/18/commits/93fbd2eeef1239615909858f4e7a1700bc0c75ca). I've released a new version `0.4.1-dev.20230703082439` from #23.